src/main/java/com/rackspace/ceres/app/services/DataTablesStatements.java

```
@@ -69,74 +73,72 @@ String partitionTableNameSuffix(Duration width) {
-                  TENANT, TIME_PARTITION_SLOT, SERIES_SET_HASH, AGGREGATOR, TIMESTAMP, VALUE)
-                  + ")"
-                  + " VALUES (?, ?, ?, ?, ?, ?)"
-          );
-
-          downsampleQueries.put(granularity.getWidth(),
-              "SELECT " + String.join(",", TIMESTAMP, VALUE)
-                  + " FROM " + tableNameDownsampled(granularity.getWidth(),
-                  granularity.getPartitionWidth())
-                  + " WHERE " + TENANT + " = ?"
-                  + "  AND " + TIME_PARTITION_SLOT + " = ?"
-                  + "  AND " + SERIES_SET_HASH + " = ?"
-                  + "  AND " + AGGREGATOR + " = ?"
-                  + "  AND " + TIMESTAMP + " >= ? AND " + TIMESTAMP + " < ?"
-          );
-
-          downsampleDeletes.put(granularity.getWidth(),
-              "DELETE FROM " + tableNameDownsampled(granularity.getWidth(),
-                  granularity.getPartitionWidth())
-                  + " WHERE " + TENANT + " = ?"
-                  + "  AND " + TIME_PARTITION_SLOT + " = ?"
-          );
-
-          downsampleDeletesWithSeriesSetHash.put(granularity.getWidth(),
-              "DELETE FROM " + tableNameDownsampled(granularity.getWidth(),
-                  granularity.getPartitionWidth())
-                  + " WHERE " + TENANT + " = ?"
-                  + "  AND " + TIME_PARTITION_SLOT + " = ?"
-                  + " AND series_set_hash = ?"
-          );
+          try {
+            downsampleInserts.put(granularity.getWidth(),
+                String
+                    .format(SpringResourceUtils.readContent("cql-queries/downsample_insert_query.cql"),
+                        tableNameDownsampled(granularity.getWidth(),
+                            granularity.getPartitionWidth()),
+                        String.join(",", TENANT, TIME_PARTITION_SLOT, SERIES_SET_HASH, AGGREGATOR,
+                            TIMESTAMP, VALUE)));
+          } catch (IOException e) {
+            e.printStackTrace();
```

Is this the correct handling? Why not let it propagate so we abandon the request entirely?

---

as its thrown from inside a lambda expression, so I can transform this exception into a new one and throws onto its caller method, so that it will fail at DataTablesStatements initialisation time.